core/runtime.rs

```diff
@@ -1325,9 +1318,12 @@ impl JsRuntime {
     self.mod_instantiate(root_id).map(|_| root_id)
   }
 
-  fn poll_pending_ops(&mut self, cx: &mut Context) -> Vec<(OpId, Box<[u8]>)> {
+  fn poll_pending_ops(
+    &mut self,
+    cx: &mut Context,
+  ) -> Vec<(PromiseId, OpResponse)> {
     let state_rc = Self::state(self.v8_isolate());
-    let mut overflow_response: Vec<(OpId, Box<[u8]>)> = Vec::new();
+    let mut overflow_response: Vec<(PromiseId, OpResponse)> = Vec::new();
```

Comment:
Is it the case that this "overflow_response" is the main (only?) mechanism now for async op responses? If so perhaps it should have different name...

Reply:
Yep, I had called it `finished_responses` in a previous PR, `ready_responses` could also be a good name.

I didn't change that here so it was easier to review/relate to the old code and understand that everything now "overflows".

Happy to change it

Reply:
Maybe "async_responses" ?

Reply:
There are a few other naming changes that would make sense in the context of this op-layer, for example it might make sense to rename `send/recv` since I intend the the op-layer to be thought of as function calls rather than request/responses with custom user payloads.

`send` could be renamed to `opcall` (similar to `syscall`) or `call_op`

Reply:
+1 for `async_responses`